### PR TITLE
Fix Global Warnings Suppression

### LINKED ISSUE #8: Global Warnings Suppression
When using `torch-pitch-shift`, all warnings are completely disabled here:
https://github.com/KentoNishi/torch-pitch-shift/blob/04cf7afd6b2d161063f115ba068b1f68f3eedc50/torch_pitch_shift/main.py#L16

By using `warnings.simplefilter("ignore")` without any scope limitation, all warnings from any imported libraries or subsequent code execution (after importing `torch-pitch-shift`) are completely disabled.
This could suppress important warnings from other libraries or tools that rely on the package and make debugging difficult for users.

Is this an intended behavior or could the suppression be scoped?

### PR DESCRIPTION
**Pull request by ramppdev**
Closes #8 by removing the hardcoded `warnings.simplefilter("ignore")` here:
https://github.com/KentoNishi/torch-pitch-shift/blob/04cf7afd6b2d161063f115ba068b1f68f3eedc50/torch_pitch_shift/main.py#L16

I have verified and tested the changes under torch 1.7 (minimum requirement), 1.13, and 2.4 (latest).

Removing the global warning suppression, there are several points of consideration inside the repo:
- Changed the dimension expansion to use numpy in `example.py`. This was a masked warning, as torch seems to not like numpy arrays wrapped in Python lists.
- Both `torch.stft` and `torch.istft` will soon require a window. This is still optional, but gives a warning for torch 2.x. As the user may want to provide an optional `window` attribute to the function, i have added this. If `window` is `None`, `torch.ones` is used, aligning with the default behavior of all torch versions (see their docs), however explicitly setting it gets rid of the warning.

Packages that rely on `torch-pitch-shift`, like [torch-audiomentations](https://github.com/asteroid-team/torch-audiomentations) may be impacted by this change and users could potentially see warnings that were previously masked. However, i would argue that this tradeoff is better than keeping it hardcoded.

Let me know what you think! @KentoNishi

### CODE REVIEW

**Review by KentoNishi — APPROVED** [review]

**Comment by KentoNishi** [comment]
tysm!
